### PR TITLE
improves instadeploy responses being swallowed by response.raise_for_…

### PR DESCRIPTION
**Pull request by emirozer**
…status call

### CODE REVIEW

**Comment by emirozer** [comment]
If the MxBuild Api return 4xx or 5xx we just discarded them silently with response.raise_for_status
this results in a bad experience of the instadeploy users. If their model is inconsistent or their java code doesnt compile they only see HTTPException  with 4xx client error or 5xx internal error. This is not helpful

File: lib/instadeploy.py [hunk 1]
```diff
@@ -169,3 +187,13 @@ def build():
         ))
 
     return response.json()
+
+
+def extract_mxbuild_response(mxbuild_response):
+    r = {}
+    if mxbuild_response and 'problems' in mxbuild_response:
+        r['buildstatus'] = json.dumps(mxbuild_response['problems'])
+        if 'message' in mxbuild_response:
```
Comment (nobeh) [inline lib/instadeploy.py]: I think this should have the same check and at the same level as `problems` and not nested inside:

```python
if mxbuild_response and 'message' in mxbuild_response:
```

Comment (emirozer) [inline lib/instadeploy.py]: :+1:

Comment (emirozer) [inline lib/instadeploy.py]: I also removed mxbuild_response None check as this is always going to be populated in the scope of this function

**Pull request merged**